### PR TITLE
Abstract out queue reading

### PR DESCRIPTION
**Pull request by psytale**

### CODE REVIEW

File: ex_py_commons/queue_reader.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import logging
+from ex_py_commons.sqs import AsyncQueue
+import asyncio
+
+logger = logging.getLogger(__name__)
+
+
+class QueueReader:
```
Comment (rymir) [inline ex_py_commons/queue_reader.py]: AsyncQueueReader?

File: ex_py_commons/queue_reader.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        messages = await self._requests.receive_messages(num_messages=10)
+        tasks = [
+            self._loop.create_task(self._process_message(message))
+            for message in messages
+        ]
+        await asyncio.wait(tasks)
+        log_exceptions_from_tasks(tasks)
+
+
+def log_exceptions_from_tasks(tasks):
```
Comment (rymir) [inline ex_py_commons/queue_reader.py]: can we move this into a lib?

**Comment by rymir** [comment]
can you please add one more commit to this pr moving `class RequestsMock(AsyncQueue):` to a test lib so we can import/use it in the other repos?

**Comment by psytale** [comment]
Done

**Pull request merged**